### PR TITLE
[Docs]Using OAuth2 as REST API authentication

### PR DESCRIPTION
**Pull request by withshubh**

### CODE REVIEW

File: docs/docs/how-to/oauth2-authorization.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Let's follow the steps to authorize ToolJet to access your Google profile data:
+
+- Select **add datasource** from the left sidebar, and choose **REST API** from the dialog window.
+
+:::info
+You can rename the datasource by clicking on its default name `REST API`
+:::
+
+- In the **URL** field, enter the base URL `[https://www.googleapis.com/oauth2/v1/userinfo](https://www.googleapis.com/oauth2/v1/userinfo)`; the base URL specifies the network address of the API service.
```
Comment (arpitnath) [inline docs/docs/how-to/oauth2-authorization.md:32]: we can just mention the URL right?
f.e.
`https://www.googleapis.com/oauth2/v1/userinfo`

File: docs/docs/how-to/oauth2-authorization.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+- Select authentication type as `OAuth 2.0`
+- Keep the default values for **Grant Type**, **Add Access Token To**, and **Header Prefix** i.e. `Authorization Code`, `Request Header`, and `Bearer` respectively.
+- Enter **Access Token URL**: `https://oauth2.googleapis.com/token`; this token allows users to verify their identity, and in return, receive a unique access token.
+- Enter the **Client ID** and **Client Secret** that we generated from the [Google Console](http://console.developers.google.com/).
+- In the **Scope** field, enter `[https://www.googleapis.com/auth/userinfo.profile](https://www.googleapis.com/auth/userinfo.profile)`; Scope is a mechanism in OAuth 2.0 to limit an application's access to a user's account. Check the scopes available for [Google OAuth2 API here](https://developers.google.com/identity/protocols/oauth2/scopes#oauth2).
```
Comment (arpitnath) [inline docs/docs/how-to/oauth2-authorization.md:37]: same here, `https://www.googleapis.com/auth/userinfo.profile`

File: docs/docs/how-to/oauth2-authorization.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- Enter **Authorization URL:** `[https://accounts.google.com/o/oauth2/v2/auth](https://accounts.google.com/o/oauth2/v2/auth)` ; the Authorization URL requests authorization from the user.
```
Comment (arpitnath) [inline docs/docs/how-to/oauth2-authorization.md:38]: also here, `https://accounts.google.com/o/oauth2/v2/auth`

and can add "redirects to retrieve an authorization code."

File: docs/docs/how-to/oauth2-authorization.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+- Create three **Custom Authentication Parameters:**
+    - response_type: code (`code` refers to the Authorization Code)
+    - client_id: (Enter the Client ID generated in Google Console)
+    - redirect_uri: `[http://localhost:8082/oauth2/authorize](http://localhost:8082/oauth2/authorize)` if using ToolJet locally or enter this `https://app.tooljet.io/oauth2/authorize` if using ToolJet Cloud.
```
Comment (arpitnath) [inline docs/docs/how-to/oauth2-authorization.md:42]: `http://localhost:8082/oauth2/authorize`

Comment (arpitnath) [reply docs/docs/how-to/oauth2-authorization.md:42]: also for the `redirect_uri`

**Review by arpitnath — CHANGES_REQUESTED** [review]

File: docs/docs/how-to/oauth2-authorization.md (hunk 4 above)
Comment (arpitnath) [reply docs/docs/how-to/oauth2-authorization.md:42]: using a table for custom params will be better.
```md
    | params      | description |
    | ----------- | ----------- |
    | response_type | code ( `code` refers to the Authorization Code) |
    | client_id | **Client ID**  |
    | redirect_uri | `http://localhost:8082/oauth2/authorize` if using ToolJet locally or enter this `https://app.tooljet.io/oauth2/authorize` if using ToolJet Cloud.  |
```

File: docs/docs/how-to/oauth2-authorization.md (hunk 1 above)
Comment (withshubh) [reply docs/docs/how-to/oauth2-authorization.md:32]:
```suggestion
- In the **URL** field, enter the base URL `https://www.googleapis.com/oauth2/v1/userinfo`; the base URL specifies the network address of the API service.
```

File: docs/docs/how-to/oauth2-authorization.md (hunk 2 above)
Comment (withshubh) [reply docs/docs/how-to/oauth2-authorization.md:37]:
```suggestion
- In the **Scope** field, enter `https://www.googleapis.com/auth/userinfo.profile`; Scope is a mechanism in OAuth 2.0 to limit an application's access to a user's account. Check the scopes available for [Google OAuth2 API here](https://developers.google.com/identity/protocols/oauth2/scopes#oauth2).
```

File: docs/docs/how-to/oauth2-authorization.md (hunk 3 above)
Comment (withshubh) [reply docs/docs/how-to/oauth2-authorization.md:38]:
```suggestion
- Enter **Authorization URL:** `https://accounts.google.com/o/oauth2/v2/auth`; the Authorization URL requests authorization from the user and redirects to retrieve an authorization code from identity server.
```

File: docs/docs/how-to/oauth2-authorization.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+- Enter the **Client ID** and **Client Secret** that we generated from the [Google Console](http://console.developers.google.com/).
+- In the **Scope** field, enter `https://www.googleapis.com/auth/userinfo.profile`; Scope is a mechanism in OAuth 2.0 to limit an application's access to a user's account. Check the scopes available for [Google OAuth2 API here](https://developers.google.com/identity/protocols/oauth2/scopes#oauth2).
+- Enter **Authorization URL:** `https://accounts.google.com/o/oauth2/v2/auth`; the Authorization URL requests authorization from the user and redirects to retrieve an authorization code from identity server.
+- Create three **Custom Authentication Parameters:**
+    - response_type: code (`code` refers to the Authorization Code)
+    - client_id: (Enter the Client ID generated in Google Console)
+    - redirect_uri: `[http://localhost:8082/oauth2/authorize](http://localhost:8082/oauth2/authorize)` if using ToolJet locally or enter this `https://app.tooljet.io/oauth2/authorize` if using ToolJet Cloud.
```
Comment (withshubh) [inline docs/docs/how-to/oauth2-authorization.md:42]:
```suggestion
- Enter **Authorization URL:** `https://accounts.google.com/o/oauth2/v2/auth`; the Authorization URL requests authorization from the user and redirects to retrieve an authorization code from identity server.
- Create three **Custom Authentication Parameters:**
    | params      | description |
    | ----------- | ----------- |
    | response_type | code ( `code` refers to the Authorization Code) |
    | client_id | **Client ID**  |
    | redirect_uri | `http://localhost:8082/oauth2/authorize` if using ToolJet locally or enter this `https://app.tooljet.io/oauth2/authorize` if using ToolJet Cloud.  |
```

**Comment by withshubh** [comment]
@arpitnath Thanks for the review. Made the changes!
Those URLs were messed by the Notion 😓

File: docs/docs/how-to/oauth2-authorization.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+- Select authentication type as `OAuth 2.0`
+- Keep the default values for **Grant Type**, **Add Access Token To**, and **Header Prefix** i.e. `Authorization Code`, `Request Header`, and `Bearer` respectively.
+- Enter **Access Token URL**: `https://oauth2.googleapis.com/token`; this token allows users to verify their identity, and in return, receive a unique access token.
+- Enter the **Client ID** and **Client Secret** that we generated from the [Google Console](http://console.developers.google.com/).
+- In the **Scope** field, enter `https://www.googleapis.com/auth/userinfo.profile`; Scope is a mechanism in OAuth 2.0 to limit an application's access to a user's account. Check the scopes available for [Google OAuth2 API here](https://developers.google.com/identity/protocols/oauth2/scopes#oauth2).
+- Enter **Authorization URL:** `https://accounts.google.com/o/oauth2/v2/auth`; the Authorization URL requests authorization from the user and redirects to retrieve an authorization code from identity server.
+- Create three **Custom Authentication Parameters:**
+    | params      | description |
```
Comment (arpitnath) [inline docs/docs/how-to/oauth2-authorization.md:40]: a space is required here

**Review by arpitnath — CHANGES_REQUESTED** [review]

File: docs/docs/how-to/oauth2-authorization.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    | ----------- | ----------- |
+    | response_type | code ( `code` refers to the Authorization Code) |
+    | client_id | **Client ID**  |
+    | redirect_uri | `http://localhost:8082/oauth2/authorize` if using ToolJet locally or enter this `https://app.tooljet.io/oauth2/authorize` if using ToolJet Cloud.  |
+- Keep the default selection for **Client Authentication** and **Save** the datasource.
+
+<img class="screenshot-full" src="/img/how-to/oauth2-authorization/restapi.png" alt="ToolJet - How To - REST API authentication using OAuth 2.0"/>
```
Comment (withshubh) [inline docs/docs/how-to/oauth2-authorization.md:47]:
```suggestion
- Enter **Authorization URL:** `https://accounts.google.com/o/oauth2/v2/auth`; the Authorization URL requests authorization from the user and redirects to retrieve an authorization code from identity server.
- Create three **Custom Authentication Parameters:**

    | params      | description |
    | ----------- | ----------- |
    | response_type | code ( `code` refers to the Authorization Code) |
    | client_id | **Client ID**  |
    | redirect_uri | `http://localhost:8082/oauth2/authorize` if using ToolJet locally or enter this `https://app.tooljet.io/oauth2/authorize` if using ToolJet Cloud.  |
    
- Keep the default selection for **Client Authentication** and **Save** the datasource.

<img class="screenshot-full" src="/img/how-to/oauth2-authorization/restapi.png" alt="ToolJet - How To - REST API authentication using OAuth 2.0"/>
```

**Review by arpitnath — APPROVED** [review]

**Pull request merged**